src/library/scala/collection/IterableOnce.scala

```diff
@@ -743,56 +765,91 @@ trait IterableOnceOps[+A, +CC[_], +C] extends Any { this: IterableOnce[A] =>
+   *                 computation of the result; must be a neutral element for `op` for the
+   *                 result to always be the same across runs.
+   *  @param op      A binary operator; must be associative for the result to always be the
+   *                 same across runs.
+   *  @return        The result of applying `op` between all the elements and `z`, or `z`
+   *                 if this $coll is empty.
    */
   def fold[A1 >: A](z: A1)(op: (A1, A1) => A1): A1 = foldLeft(z)(op)
 
-  /** Reduces the elements of this $coll using the specified associative binary operator.
+  /** Applies the given binary operator to all elements of this $coll.
    *
-   *  $undefinedorder
+   *  For each application of the operator, each operand is either an element of this
+   *  $coll or another such application of the operator. $undefinedOrder $exactlyOnce
    *
-   *  @tparam B      A type parameter for the binary operator, a supertype of `A`.
-   *  @param op       A binary operator that must be associative.
-   *  @return         The result of applying reduce operator `op` between all the elements if the $coll is nonempty.
-   *  @throws UnsupportedOperationException if this $coll is empty.
+   *  If this collection is ordered, then for any application of the operator, the
+   *  element(s) appearing in the left operand will precede those in the right.
+   *
+   *  $orderIndependentReduce
+   *  $willNotTerminateInf
+   *
+   *  @tparam B      The type parameter for the binary operator, a supertype of `A`.
+   *  @param op      A binary operator; must be associative for the result to always be the
+   *                 same across runs.
+   *  @return        The result of applying `op` between all the elements if the $coll is
+   *                 nonempty.
+   *  @throws        UnsupportedOperationException if this $coll is empty.
    */
   def reduce[B >: A](op: (B, B) => B): B = reduceLeft(op)
 
-  /** Reduces the elements of this $coll, if any, using the specified
-   *  associative binary operator.
+  /** Optionally reduces this $coll with the given binary operator.
+   *
+   *  The behavior is the same as [[reduce]] except that the value is `None` if the $coll
+   *  is empty. $undefinedOrder $exactlyOnce
    *
-   *  $undefinedorder
+   *  $orderIndependentReduce
+   *  $willNotTerminateInf
    *
-   *  @tparam B     A type parameter for the binary operator, a supertype of `A`.
-   *  @param op      A binary operator that must be associative.
-   *  @return        An option value containing result of applying reduce operator `op` between all
-   *                 the elements if the collection is nonempty, and `None` otherwise.
+   *  @tparam B      A type parameter for the binary operator, a supertype of `A`.
+   *  @param op      A binary operator; must be associative for the result to always be the
+   *                 same across runs.
+   *  @return        An option value containing the result of reducing this $coll with `op`
```

Comment:
the wording feels awkward here to me; I think using `Some` would help

Reply:
I would like to help but everything is awkward tonight. Obviously, to use the adverbial disjunct preferred by math types,
```
Optionally, the result of reducing this $coll with `op`.
```
In academia, a disjunct is when an adjunct is laid off after teaching six courses a semester for peanuts.

Reply:
Tried different wording in latest commit, does it sound better?